src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,154 @@
+            storeFrameworkId(fwkId);
+        } catch (StateStoreException ex) {
+            log.error("Failed to set FrameworkID: " + fwkId, ex);
+            throw ex;
+        }
+    }
+
+    public void recordTasks(List<TaskInfo> taskInfos) throws StateStoreException {
+        log.info(String.format("Recording %d updated TaskInfos/TaskStatuses:", taskInfos.size()));
+        List<TaskStatus> taskStatuses = new ArrayList<>();
+        for (TaskInfo taskInfo : taskInfos) {
+            TaskStatus taskStatus = TaskStatus.newBuilder()
+                    .setTaskId(taskInfo.getTaskId())
+                    .setExecutorId(taskInfo.getExecutor().getExecutorId())
+                    .setState(TaskState.TASK_STAGING)
+                    .build();
+            log.info(String.format("- %s => %s", taskInfo, taskStatus));
+            log.info("Marking stopped task as failed: {}", TextFormat.shortDebugString(taskInfo));
+            taskStatuses.add(taskStatus);
+        }
+
+        storeTasks(taskInfos);
+        for (TaskStatus taskStatus : taskStatuses) {
+            recordTaskStatus(taskStatus);
+        }
+    }
+
+    public void updateStatus(TaskStatus taskStatus) throws StateStoreException {
+        log.info(String.format("Recording updated TaskStatus to state store: %s", taskStatus));
+        recordTaskStatus(taskStatus);
+    }
+
+    public List<TaskInfo> getTerminatedTaskInfos() throws Exception {
+        return new ArrayList(fetchTerminatedTasks());
+    }
+
+    public int getRunningBrokersCount() throws StateStoreException {
+        int count = 0;
+
+        for (TaskStatus taskStatus : getTaskStatuses()) {
+            if (taskStatus.getState().equals(TaskState.TASK_RUNNING)) {
+                count++;
+            }
+        }
+
+        return count;
+    }
+
+    @Override
+    public Set<TaskStatus> getTaskStatuses() throws StateStoreException {
+        Set<TaskStatus> taskStatuses = new HashSet<TaskStatus>();
+        taskStatuses.addAll(fetchStatuses());
+        return taskStatuses;
+    }
+
+    public List<TaskInfo> getTaskInfos() throws StateStoreException {
+        List<TaskInfo> taskInfos = new ArrayList<TaskInfo>();
+        taskInfos.addAll(fetchTasks());
+        return taskInfos;
+    }
+
+    public List<Resource> getExpectedResources() {
+        List<Resource> resources = new ArrayList<>();
+        try {
+            for (TaskInfo taskInfo : getTaskInfos()) {
+                resources.addAll(taskInfo.getResourcesList());
+                if (taskInfo.hasExecutor()) {
+                    resources.addAll(taskInfo.getExecutor().getResourcesList());
+                }
+            }
+        } catch (Exception ex) {
+            log.error("Failed to retrieve all Task information", ex);
+            return resources;
+        }
+        return resources;
+    }
+
+    public void recordTaskInfo(TaskInfo taskInfo) throws StateStoreException {
```

Comment:
Well, since I'm moving from inheritance to composition, we have to wrap this in order to expose it, unless you want me to expose the underlying `StateStore`